### PR TITLE
Feature/webview for model view

### LINKED ISSUE #1454: support webview component in view model
- SQL Operations Studio Version:

Steps to Reproduce:

### PR DESCRIPTION
**Pull request by AbbiePetcht**
Fix  #1454: support webview component in view model

### CODE REVIEW

File: src/sql/parts/modelComponents/webview.component.ts [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+import * as sqlops from 'sqlops';
+import Event, { Emitter } from 'vs/base/common/event';
+import { Webview } from 'vs/workbench/parts/html/browser/webview';
+import { addDisposableListener, EventType } from 'vs/base/browser/dom';
+import { IDisposable, Disposable } from 'vs/base/common/lifecycle';
+import { IPartService, Parts } from 'vs/workbench/services/part/common/partService';
+
+import { ComponentBase } from 'sql/parts/modelComponents/componentBase';
+import { IComponent, IComponentDescriptor, IModelStore, ComponentEventType } from 'sql/parts/modelComponents/interfaces';
+import { Dropdown, IDropdownOptions } from 'sql/base/browser/ui/editableDropdown/dropdown';
```
Comment (llali) [inline src/sql/parts/modelComponents/webview.component.ts]: remove this

File: src/sql/parts/modelComponents/webview.component.ts [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+import { CommonServiceInterface } from 'sql/services/common/commonServiceInterface.service';
+import { attachListStyler } from 'vs/platform/theme/common/styler';
+import { attachEditableDropdownStyler } from 'sql/common/theme/styler';
```
Comment (llali) [inline src/sql/parts/modelComponents/webview.component.ts]: also you don't need the styler

File: src/sql/parts/modelComponents/webview.component.ts [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+		this.baseDestroy();
+	}
+
+	/// Webview Functions
+
+	private setHtml(): void {
+		if (this._webview && this.html) {
+			this._renderedHtml = this.html;
+			this._webview.contents = this._renderedHtml;
+			this._webview.layout();
```
Comment (llali) [inline src/sql/parts/modelComponents/webview.component.ts]: are you sure you need to call layout her? maybe calling this method in layput method is enough?

File: src/sql/parts/modelComponents/webview.component.ts [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+	}
+
+	public setLayout(layout: any): void {
+		// TODO allow configuring the look and feel
+		this.layout();
+	}
+
+	public setProperties(properties: { [key: string]: any; }): void {
+		super.setProperties(properties);
+		if (this.html !== this._renderedHtml) {
```
Comment (llali) [inline src/sql/parts/modelComponents/webview.component.ts]: do you need to compare the strings? or is this enough?

Comment (AbbiePetcht) [reply src/sql/parts/modelComponents/webview.component.ts]: It is enough

---
In reply to: [190075853](https://github.com/Microsoft/sqlopsstudio/pull/1463#discussion_r190075853) [](ancestors = 190075853)

File: src/sql/parts/modelComponents/webview.component.ts (hunk 3 above)
Comment (AbbiePetcht) [reply src/sql/parts/modelComponents/webview.component.ts]: I followed webview in dashboard. It calls layout everytime it html is set

---
In reply to: [190075588](https://github.com/Microsoft/sqlopsstudio/pull/1463#discussion_r190075588) [](ancestors = 190075588)

**Pull request merged**